src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java

```diff
@@ -202,6 +197,19 @@ private SscsCaseDetails createCaseInCcd(SscsCaseData caseData, EventType eventTy
                     caseDetails != null ? caseDetails.getId() : "", caseData.getGeneratedNino(),
                     caseData.getAppeal().getBenefitType().getCode(), e.getMessage()), e);
         }
+
+        if (caseDetails != null) {
+            log.info("Duplicate case {} found for Nino {} and benefit type {}. "
+                            + "No need to continue with post create case processing.",
+                    caseDetails.getId(), caseData.getGeneratedNino(),
+                    caseData.getAppeal().getBenefitType().getCode());
+            throw new DuplicateCaseException(
+                    String.format("An appeal has already been submitted, for that decision date %s \n "
```

Comment:
It might read better if the bulk of the string message was put into the DuplicateCaseException class